src/main/java/org/opensearch/neuralsearch/processor/normalization/MinMaxScoreNormalizationTechnique.java

```diff
@@ -167,22 +225,182 @@ private float[] getMinScores(final List<CompoundTopDocs> queryTopDocs, final int
         return minScores;
     }
 
-    private float normalizeSingleScore(final float score, final float minScore, final float maxScore) {
+    private float normalizeSingleScore(final float score, final float minScore, final float maxScore, final LowerBound lowerBound) {
         // edge case when there is only one score and min and max scores are same
         if (Floats.compare(maxScore, minScore) == 0 && Floats.compare(maxScore, score) == 0) {
             return SINGLE_RESULT_SCORE;
         }
-        float normalizedScore = (score - minScore) / (maxScore - minScore);
-        return normalizedScore == 0.0f ? MIN_SCORE : normalizedScore;
+        if (!lowerBound.isEnabled()) {
+            return LowerBound.Mode.IGNORE.normalize(score, minScore, maxScore, lowerBound.getMinScore());
+        }
+        return lowerBound.getMode().normalize(score, minScore, maxScore, lowerBound.getMinScore());
+    }
+
+    private Optional<List<Pair<LowerBound.Mode, Float>>> getLowerBounds(final Map<String, Object> params) {
```

Comment:
Please add comments on what is happening in this method to better understand.

Reply:
ack